bcda/auth/backend_test.go

```diff
@@ -123,13 +125,52 @@ func (s *BackendTestSuite) TestInitAuthBackend() {
 	assert.IsType(s.T(), &rsa.PublicKey{}, s.AuthBackend.PublicKey)
 }
 
-func (s *BackendTestSuite) TestHashCompare() {
+func (s *BackendTestSuite) TestHashComparable() {
 	uuidString := uuid.NewRandom().String()
-	hash := auth.NewHash(uuidString)
+	hash, err := auth.NewHash(uuidString)
+	assert.Nil(s.T(), err)
 	assert.True(s.T(), hash.IsHashOf(uuidString))
 	assert.False(s.T(), hash.IsHashOf(uuid.NewRandom().String()))
 }
 
+func (s *BackendTestSuite) TestHashUnique() {
+	uuidString := uuid.NewRandom().String()
+	hash1, _ := auth.NewHash(uuidString)
+	hash2, _ := auth.NewHash(uuidString)
+	assert.NotEqual(s.T(), hash1.String(), hash2.String())
+}
+
+func (s *BackendTestSuite) TestHashTime() {
+	uuidString := uuid.NewRandom().String()
+
+	start := time.Now()
+	hash, _ := auth.NewHash(uuidString)
+	hashCreationTime := time.Since(start)
+
+	start = time.Now()
+	hash.IsHashOf(uuidString)
+	hashComparisonTime := time.Since(start)
+
+	// Intentionally printing actual execution times even when within parameters
+	fmt.Printf("Hash creation took %s\nHash comparison took %s\n", hashCreationTime, hashComparisonTime)
+	// Lower limit has some padding.  Current number of iterations is tuned to take about 1s on a 2.8 GHz Intel Core i7
+	assert.True(s.T(), hashComparisonTime > 500*time.Millisecond, "Hash comparison took %s; target time > .5s")
+	// Upper limit is generous, so that there's no trouble passing tests on slower virtual environments.
+	assert.True(s.T(), hashComparisonTime < 5*time.Second, "Hash comparison took %s, target time < 5s")
+}
+
+func (s *BackendTestSuite) TestHashCompatibility() {
+	uuidString := "811ab5dd-df67-4f3a-88bf-46ed91efbf73"
+	hash := auth.Hash("RFbxyt3Fa/zQTZ324QbgTRv/yj5cXdaq5CdVDBvliN8=:8c7EH5a/I76UFgwO0F8GRTD1pfqDvrbrwN5qbuvKUZO90BuBm4WQFQynhnKE1gtvmF5xJlyoEPeqnK44TfSMuQ==")
+	assert.True(s.T(), hash.IsHashOf(uuidString), "Possible change in hashing parameters or algorithm.  Known input/output does not match.  Merging this code will result in invalidating credentials.")
+}
+
+func (s *BackendTestSuite) TestHashEmpty() {
+	hash, err := auth.NewHash("")
+	assert.Nil(s.T(), err)
+	assert.False(s.T(), hash.IsHashOf(""))
+}
+
```

Comment:
Can we get a test for an invalid source hash?  For example "RFbxyt3Fa/zQTZ324QbgTRv/yj5cXdaq5CdVDBvliN8=:8c7EH5a/I76UFgwO0F8GRTD1pfqDvrbrwN5qbuvKUZO90BuB:m4WQFQynhnKE1gtvmF5xJlyoEPeqnK44TfSMuQ==" will fail because it has 3 sections instead fo 2 when split on the colon.  sb an easy way to bump up that coverage number.